### PR TITLE
feat(sync-repo-settings): allow configuring extra team permissions by language

### PR DESCRIPTION
**Pull request by chingor13**
Adds a new teams.json source file to contain the additional teams.

Alternatively, we can add the teams configuration in the `required-checks.json` which we should probably rename to `settings.json` as it's grown beyond just required status checks.

### CODE REVIEW

File: packages/sync-repo-settings/package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "probot-app"
   ],
   "scripts": {
-    "compile": "tsc -p . && cp ./src/required-checks.json ./build/src/required-checks.json.json",
+    "compile": "tsc -p . && cp ./src/required-checks.json ./build/src/required-checks.json.json && cp ./src/teams.json ./build/src/teams.json.json",
```
Comment (JustinBeckwith) [inline packages/sync-repo-settings/package.json:21]: Why do we need to copy these json files at all into the build?  We could include them in the module via the `files` field in `package.json`, and rewrite the paths so we don't need these gymnastics.

File: packages/sync-repo-settings/src/sync-repo-settings.ts [hunk 2]
```diff
@@ -35,6 +37,17 @@ interface LanguageConfig {
   };
 }
 
+type Permission = 'pull' | 'push' | 'admin' | 'maintain' | 'triage';
```
Comment (JustinBeckwith) [inline packages/sync-repo-settings/src/sync-repo-settings.ts:40]: I'm fine having this here, but I also know this is expressed somewhere in OctoKit already :)

File: packages/sync-repo-settings/src/teams.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+    "java": [
+        {
+            "slug": "java-samples-reviewers",
```
Comment (JustinBeckwith) [inline packages/sync-repo-settings/src/teams.json:4]: Per our discussion in the team meeting today, I love the idea of this being generalizable across languages.

File: packages/sync-repo-settings/src/sync-repo-settings.ts [hunk 4]
```diff
@@ -154,30 +167,43 @@ handler.updateMasterBranchProtection = async function updateMasterBranchProtecti
...
       permission: 'push',
     },
   ];
+};
+
+handler.getRepoTeams = function (repo: Repo): TeamPermission[] {
+  const language = repo.language;
+  let teams = handler.defaultLanguageTeams(language);
+  if (language in languageTeams) {
+    teams = teams.concat(languageTeams[language]);
```
Comment (JustinBeckwith) [inline packages/sync-repo-settings/src/sync-repo-settings.ts:191]: You could keep `teams` a `const`, and do `teams.push(...languageTeams[language])` as well.  (non-blocking feedback)

File: packages/sync-repo-settings/src/teams.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+            "permission": "push"
+        }
+    ]
+}
```
Comment (JustinBeckwith) [inline packages/sync-repo-settings/src/teams.json:8]: Can I get a newline?

File: packages/sync-repo-settings/src/sync-repo-settings.ts (hunk 2 above)
Comment (chingor13) [reply packages/sync-repo-settings/src/sync-repo-settings.ts:40]: Looks like octokit does not publish a type or enum for these constants

File: packages/sync-repo-settings/package.json (hunk 1 above)
Comment (chingor13) [reply packages/sync-repo-settings/package.json:21]: I'm not sure why this is here, but I followed the pattern.

File: packages/sync-repo-settings/src/sync-repo-settings.ts (hunk 4 above)
Comment (chingor13) [reply packages/sync-repo-settings/src/sync-repo-settings.ts]: Done

File: packages/sync-repo-settings/src/teams.json (hunk 5 above)
Comment (chingor13) [reply packages/sync-repo-settings/src/teams.json]: Done

File: packages/sync-repo-settings/package.json (hunk 1 above)
Comment (chingor13) [reply packages/sync-repo-settings/package.json:21]: Filed https://github.com/googleapis/repo-automation-bots/issues/603 to track. I'm not sure how to fix this.

File: packages/sync-repo-settings/src/teams.json (hunk 3 above)
Comment (chingor13) [reply packages/sync-repo-settings/src/teams.json:4]: For now, I think only java-samples-reviewers exists and we would like this group on all the java repos. We can refactor this one the internal design is approved.

**Pull request merged**